intra/doh/doh.go

```diff
@@ -299,10 +299,8 @@ func (t *transport) doQuery(q []byte) (response []byte, server *net.TCPAddr, qer
 		}
 
 		response = tryServfail(q)
-	}
-
-	if server != nil {
-		// Record a working IP address for this server
+	} else if server != nil {
```

Comment:
Could also move this _confirm IP_ block to the `defer`ed function in [`doh:transport.sendRequest`](https://github.com/Jigsaw-Code/outline-go-tun2socks/blob/a0d38451/intra/doh/doh.go#L321-L344), or alternatively, move the block which _disconfirms IP_ to here?

Reply:
That's possible, but it would result in slightly different behavior.  I would prefer to put it here, so we only confirm after all checks have completed (including the query ID check [here](https://github.com/Jigsaw-Code/outline-go-tun2socks/pull/80/files#diff-355c4ef8052beef46dc570a6de847f873c25583bc92b38fe9f382a9e398b255aR283-R291)).

Reply:
Gotcha.